### PR TITLE
Reduced complexity of HomeUtils.kt

### PR DESCRIPTION
**Pull request by Jeremmmyyyyy**
This will reduce the complexity of the HomeUtils.kt class under 50.

### CODE REVIEW

File: app/src/main/java/com/example/triptracker/view/home/HomeUtils.kt [hunk 1]
```diff
@@ -353,10 +357,7 @@ fun DisplayItinerary(
                               color = md_theme_light_onPrimary,
                               modifier = Modifier.testTag("Title"))
                           Text(
-                              text =
-                                  if (ambientProfile.userProfile.value.flowerMode == 1)
-                                      "${itinerary.flameCount} ${flowerStringBasedOnCount(itinerary.flameCount)}"
-                                  else "${itinerary.flameCount} 🔥",
+                              text = getText(ambientProfile, itinerary),
```
Comment (cleorenaud) [inline app/src/main/java/com/example/triptracker/view/home/HomeUtils.kt:360]: Maybe you could rename the function to something like getTrendingCount to make it easier to understand

File: app/src/main/java/com/example/triptracker/view/home/HomeUtils.kt [hunk 2]
```diff
@@ -373,23 +374,10 @@ fun DisplayItinerary(
                           if (displayImage) {
                             Spacer(modifier = Modifier.height(screenHeight * 0.02f))
                             LazyRow(
-                                modifier =
-                                    Modifier.height(
-                                        if (imageIsEmpty.value) 0.dp else screenHeight * 0.25f),
+                                modifier = Modifier.getHeight(imageIsEmpty, screenHeight),
```
Comment (cleorenaud) [inline app/src/main/java/com/example/triptracker/view/home/HomeUtils.kt:377]: Nice way to decrease the code complexity !

File: app/src/main/java/com/example/triptracker/view/home/HomeUtils.kt [hunk 3]
```diff
@@ -466,6 +454,53 @@ private fun ShowAlert(onDismiss: () -> Unit, onConfirm: () -> Unit) {
       })
 }
 
+/**
+ * Displays the images of the pinned places in the itinerary
+ *
+ * @param pin: Pin object to display
+ * @param imageIsEmpty: MutableState to check if the image is empty
+ */
+@Composable
```
Comment (cleorenaud) [inline app/src/main/java/com/example/triptracker/view/home/HomeUtils.kt:463]: Nice comments, thanks !

**Review by SpaceMercury — APPROVED** [review]
Nice PR! Great comments for the extracted functions!

**Review by cleorenaud — APPROVED** [review]
Everything looks good to me, thanks for taking care of this !
I just left you one small suggestion to rename one of the newly created function but otherwise good job !

File: app/src/main/java/com/example/triptracker/view/home/HomeUtils.kt (hunk 1 above)
Comment (Jeremmmyyyyy) [reply app/src/main/java/com/example/triptracker/view/home/HomeUtils.kt:360]: Done ! thanks for the comment !

**Pull request merged**